services/helper/sessions.js

```diff
@@ -1,64 +1,703 @@
+
+            return common.successResponse({
+                statusCode: httpStatusCode.created,
+                message: apiResponses.SESSION_CREATED_SUCCESSFULLY,
+                result: data
+            });
+
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    static async update(sessionId, bodyData, userId, method) {
+        let isSessionReschedule = false;
+        try {
+
+            if (!await this.verifyMentor(userId)) {
+                return common.failureResponse({
+                    message: apiResponses.INVALID_PERMISSION,
+                    statusCode: httpStatusCode.bad_request,
+                    responseCode: 'CLIENT_ERROR'
+                });
+            }
+
+            const sessionDetail = await sessionData.findSessionById(ObjectId(sessionId));
+
+            console.log(sessionDetail);
+
+            if (!sessionDetail) {
+                return common.failureResponse({
+                    message: apiResponses.SESSION_NOT_FOUND,
+                    statusCode: httpStatusCode.bad_request,
+                    responseCode: 'CLIENT_ERROR'
+                });
+            }
+
+            if (bodyData.startDate) {
+                bodyData['startDateUtc'] = moment.unix(bodyData.startDate).utc().format(common.UTC_DATE_TIME_FORMAT);
+                isSessionReschedule = true;
+            }
+            if (bodyData.endDate) {
+                bodyData['endDateUtc'] = moment.unix(bodyData.endDate).utc().format(common.UTC_DATE_TIME_FORMAT);
+                isSessionReschedule = true;
+            }
+
+            let message;
+            let updateData;
+            if (method == common.DELETE_METHOD) {
+                updateData = { deleted: true };
+                message = apiResponses.SESSION_DELETED_SUCCESSFULLY;
+
+            } else {
+
+                updateData = bodyData;
+                message = apiResponses.SESSION_UPDATED_SUCCESSFULLY;
+            }
+
+            updateData.updatedAt = new Date().getTime();
+            const result = await sessionData.updateOneSession({
+                _id: ObjectId(sessionId)
+            }, updateData);
+
+            if (result === 'SESSION_ALREADY_UPDATED') {
+                return common.failureResponse({
+                    message: apiResponses.SESSION_ALREADY_UPDATED,
+                    statusCode: httpStatusCode.bad_request,
+                    responseCode: 'CLIENT_ERROR'
+                });
+            }
+
+            if (method == common.DELETE_METHOD || isSessionReschedule) {
+                const sessionAttendees = await sessionAttendesData.findAllSessionAttendees({ sessionId: ObjectId(sessionId) });
+                const sessionAttendeesIds = [];
+                sessionAttendees.forEach(attendee => {
+                    sessionAttendeesIds.push(attendee.userId.toString());
+                });
+                const attendeesAccounts = await this.getAllAccountsDetail(sessionAttendeesIds);
```

Comment:
@rakeshSgr  This API should not be in session services as it should be in sessionAttendees helper.

Reply:
@aman-g202 Please check this